StevesSixLeggedSprinters/BangazonAPI
# Kk product type

## Status
Choose:
HOLD**

## Migrations
YES

## Description
I created the ProductType model and controller (i.e. ProductType.cs and ProductTypeController.cs).
These files comprise the ProductType table. Every product will have a productType (productType is a FK of Products. The productType will have a category name.
The controller contains methods to POST(create), PUT(edit), GET(query) and DELETE(remove) ProductType objects in the database.
I also added a ProductType data field in BangazonAPIContext.cs. This establishes relationship with other tables.

I also set up an ICollection for products in the ProductType.cs file. This will allow a query of products to return productType. Make sure that Products model and controller has been added to the master in order for it to work. 

## Todos


## Deploy Notes
Not yet deployable

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```
I have set up the following methods: GET, POST, PUT, DELETE
Follow these step by step in order to test.

git fetch --a  (fetch all branches from GitHub)
git branch -avv (display all branches)
git checkout kk-product-type (switch into my branch)
git pull origin kk-product-type (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Have Postman installed: https://www.getpostman.com/
In Postman, 'Post' (add) a couple product types to the database to start testing.

Use the DB Browser in SQL to view the data and observe outputs.

Example:
{
        "CategoryName": "Electronics"
}

To view all products:
Select GET
http://localhost:5000/api/producttype

To view a single product:
Select GET
http://localhost:5000/api/producttype/{id}

To edit a single product
Select PUT
http://localhost:5000/api/producttype/{id}

To delete a single product
Select DELETE
http://localhost:5000/api/producttype/{id}
```

## Impacted Areas in Application
List general components of the application that this PR will affect:

* Models/ProductType.cs
*Controllers/ProductTypeController.cs
*Data/BangazonAPIContext.cs

Reference link to the ticket:
https://github.com/StevesSixLeggedSprinters/BangazonAPI/issues/5
